### DISCUSSION TITLE
Does Zephyr have any conventions for node names in Devicetree?

### DISCUSSION THREAD

**Discussion by cdwilson** [zephyrproject-rtos/zephyr#67495] [Q&A] [answered]
I came across https://www.kernel.org/doc/html/next/devicetree/bindings/dts-coding-style.html#naming-and-valid-characters which mentions some Devicetree naming rules used by the Linux Kernel:

> Node and property names can use only the following characters:
> 
> 1. Lowercase characters: [a-z]
> 2. Digits: [0-9]
> 3. Dash: -
> 
> Labels can use only the following characters:
> 
> 1. Lowercase characters: [a-z]
> 2. Digits: [0-9]
> 3. Underscore: _

The examples shown in https://docs.zephyrproject.org/latest/build/dts/intro-syntax-structure.html appear to follow the rules mentioned above, but in the actual tree, there are lots of examples where it appears these rules are not followed. For example, nodes in the example below use `_` instead of `-`:

https://github.com/zephyrproject-rtos/zephyr/blob/c2f0a7a5a37dab5f5a42550e5a2b66dca987905e/dts/bindings/led/gpio-leds.yaml#L11-L24

Does Zephyr have any rules or conventions regarding the naming of nodes & node labels in Devicetree?

**Comment by ndrs-pst** [comment, marked as answer]
@cdwilson, this issue is also under discussion in the following https://github.com/zephyrproject-rtos/zephyr/issues/53506.
Based on this, I recommend aligning with the guidelines outlined in the documentation.

**Reply by cdwilson** [reply]
@ndrs-pst is that proposal limited to properties only or does it also include node names?

**Reply by ndrs-pst** [reply]
@cdwilson, from my perspective, the proposal places a primary emphasis on properties as the initial priority, leaving open the possibility of extending its application to node names in the future.
